File: Android 11 - Content providers and Broadcast receivers/ContentProviderExercise2/app/src/main/java/com/programming/android/sdu/contentproviderexercise/ContactsAdapter.java
package com.programming.android.sdu.contentproviderexercise;

import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

public class ContactsAdapter extends RecyclerView.Adapter<ContactsAdapter.ViewHolder> {
    List<ContactPerson> contacts;

    public ContactsAdapter(List<ContactPerson> numbers) {
        this.contacts = numbers;
    }


    @Override
    public ContactsAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.contact_list_item, parent, false);
        ViewHolder vh = new ViewHolder(v);
        return vh;
    }



    @Override
    public void onBindViewHolder(final ContactsAdapter.ViewHolder holder, int position) {
        ContactPerson person = contacts.get(position);
        holder.column1.setText(person.id);
        holder.column2.setText(person.displayName);
        holder.column3.setText(person.lookupKey);
    }

    @Override
    public int getItemCount() {
        return contacts.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        public TextView column1;
        public TextView column2;
        public TextView column3;


        public ViewHolder(View container) {
            super(container);
            column1 = container.findViewById(R.id.tvColumn1);
            column2 = container.findViewById(R.id.tvColumn2);
            column3 = container.findViewById(R.id.tvColumn3);
        }
    }




}

File: Android 11 - Content providers and Broadcast receivers/ContentProviderExercise2_mvvm/app/src/main/java/com/programming/android/sdu/contentproviderexercise/ContactsAdapter.kt
package com.programming.android.sdu.contentproviderexercise

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class ContactsAdapter(private var contacts: List<ContactPerson>) : RecyclerView.Adapter<ContactsAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.contact_list_item, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val person = contacts[position]
        holder.column1.text = person.id
        holder.column2.text = person.displayName
        holder.column3.text = person.lookupKey
    }

    override fun getItemCount(): Int {
        return contacts.size
    }

    class ViewHolder(container: View) : RecyclerView.ViewHolder(container) {
        var column1: TextView = container.findViewById(R.id.tvColumn1)
        var column2: TextView = container.findViewById(R.id.tvColumn2)
        var column3: TextView = container.findViewById(R.id.tvColumn3)
    }
}